ansible/roles/docker/tasks/main.yml

```diff
@@ -41,26 +42,63 @@
   shell: systemctl status docker-tcp.socket | grep "Active:"
   register: docker_tcp_socket_state
 
+- include: create_docker_device.yml
+  when: docker_device != ""
+
 - name: copy systemd units for docker tcp socket settings
   template: src=docker-tcp.j2 dest=/etc/systemd/system/docker-tcp.socket
   register: docker_tcp_socket
 
   # tcp socket service requires docker service to be started after it
-- name: start docker tcp socket service
-  shell: sudo systemctl stop docker && sudo systemctl start docker-tcp.socket && sudo systemctl start docker
-  when: (docker_tcp_socket | changed) or not docker_tcp_socket_state.stdout | match("Active.* active")
+- name: reload systemd configuration
+  shell: sudo systemctl daemon-reload
+  when: "(docker_tcp_socket | changed) or (docker_tcp_socket_state.stdout != 'Active: active')"
+
+- name: stop docker
+  service:
+    name: docker
+    state: stopped
+  when: "(docker_tcp_socket | changed) or (docker_tcp_socket_state.stdout != 'Active: active')"
+
+- name: start docker-tcp service
+  service:
+    name: docker-tcp.socket
+    state: started
+  when: "(docker_tcp_socket | changed) or (docker_tcp_socket_state.stdout != 'Active: active')"
+
+- name: check docker service state
+  shell: sudo systemctl status docker | grep 'Active.*active' -o
+  ignore_errors: true
+  register: docker_service_state
+  tags:
+    - prebake-for-dev
 
 - name: remove the docker key file, if any. It shall be regenerated by docker on restart
   file: name=/etc/docker/key.json  state=absent
```

Comment:
this might be another merge issue, this task is run conditionally in ansible/master

https://github.com/contiv/ansible/blob/master/roles/docker/tasks/main.yml#L68